shotover/src/transforms/kafka/sink_cluster/shotover_node.rs

```diff
@@ -26,16 +32,79 @@ impl ShotoverNodeConfig {
 
 #[derive(Clone)]
 pub(crate) struct ShotoverNode {
-    pub address: KafkaAddress,
+    pub address: SocketAddr,
     pub rack: StrBytes,
     pub broker_id: BrokerId,
     #[allow(unused)]
     state: Arc<AtomicShotoverNodeState>,
 }
 
+impl ShotoverNode {
+    #![allow(unused)]
+    pub(crate) fn is_up(&self) -> bool {
+        self.state.load(Ordering::Relaxed) == ShotoverNodeState::Up
+    }
+
+    pub(crate) fn set_state(&self, state: ShotoverNodeState) {
+        self.state.store(state, Ordering::Relaxed)
+    }
+}
+
 #[atomic_enum]
 #[derive(PartialEq)]
 pub(crate) enum ShotoverNodeState {
     Up,
     Down,
 }
+
+pub(crate) fn start_shotover_peers_check(
+    shotover_peers: Vec<ShotoverNode>,
+    check_shotover_peers_delay_ms: u64,
+    connect_timeout_ms: u64,
+) {
+    tokio::spawn(async move {
+        // Wait for all shotover nodes to start
+        sleep(Duration::from_secs(10)).await;
+        loop {
```

Comment:
Do we need this outer loop?

Reply:
I was thinking about adding error handling for check_shotover_peers like in https://github.com/shotover/shotover-proxy/blob/0db4f69e39eec2944c6a9141bbad7b2919613aee/shotover/src/transforms/kafka/sink_cluster/scram_over_mtls.rs#L47 so that it can be restarted if somehow fails. But since I was not sure how check_shotover_peers should return a result, the error handling was not added and this loop is not useful in this case. Do you think we should handle its error at all?

Reply:
Thats a good point.
For something like `StdRng::from_rng(rand::thread_rng()).unwrap();` we should return that error, log it, and restart the task from the begginning, like we do for scram_over_mtls